Src/Web/Web.Shared.Net/SyntheticUserAgentTelemetryInitializer.cs

```diff
@@ -49,19 +51,26 @@ protected override void OnInitializeTelemetry(HttpContext platformContext, Reque
             {
                 if (platformContext != null)
                 {
-                    var request = platformContext.GetRequest();
-
-                    if (request != null && !string.IsNullOrEmpty(request.UserAgent))
+                    if (platformContext.Items.Contains(SyntheticSourceNameKey))
```

Comment:
Current code does not seem to be reading anything from RequestTelemetry - changing that would be a behavior change instead of perf optimization attempt - do we want that for stabilization time?